execution/utility.sh

```diff
@@ -2171,26 +2172,46 @@ function manage_waf_logging() {
 
 # -- Git Repo Management --
 
+function is_git_repo() {
+  local local_dir="$1";
+
+  git -C "${local_dir}" status >/dev/null 2>&1
+}
+
+function init_git_repo() {
+  local local_dir="$1";
+
+  is_git_repo "${local_dir}" || git init "${local_dir}"
+}
+
 function in_git_repo() {
-  git status >/dev/null 2>&1
+  is_git_repo
 }
 
-function clone_git_repo() {
+function format_git_url() {
   local repo_provider="$1"; shift
   local repo_host="$1"; shift
   local repo_path="$1"; shift
+
+  if [[  (-n "${repo_provider}") &&
+      (-n "${repo_host}") &&
+      (-n "${repo_path}") ]]; then
+    local credentials_var="${repo_provider^^}_CREDENTIALS"
+    printf "https://%s@%s/%s" "${!credentials_var}" "${repo_host}" "${repo_path}"
```

Comment:
Do we need something here to gracefully fail on `${!credentials_var}` ?

Reply:
I now treat this as meaning the git repo is public and no credentials are required. The inclusion of the ampersand separator is now conditional to ensure the url is correctly formatted.